test/image_processing/lanczos_scaling.cpp

```diff
@@ -0,0 +1,82 @@
+//#include <boost/gil/image_processing/scaling.hpp>
+
+#include <boost/gil/extension/io/png.hpp>
+#include <boost/gil/extension/io/jpeg.hpp>
+#include <boost/gil/io/read_image.hpp>
+#include <boost/gil/io/write_view.hpp>
+
+#include <boost/gil/image.hpp>
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/rgb.hpp>
+#include <boost/gil/pixel.hpp>
+
+#include <boost/mpl/int.hpp>
+
+#include <cmath>
+#include <iostream>
+
+namespace gil = boost::gil;
+namespace mpl = boost::mpl;
+using image_view = decltype(gil::view(std::declval<gil::rgb8_image_t&>()));
+double sinc(double x) {
+    constexpr double pi = 3.14;
+    return std::sin(x * pi) / (x * pi);
+}
+
+double lanczos(long int x, long int a) {
+    if (x == 0) {
+        return 1;
+    }
+    if (x > -a && x < a) {
```

Comment:
`-a < x && x < a` reads clearer as range to me

Reply:
Not sure if there is a consensus on this, so leaving this open for now. @stefanseefeld , what do you think?

Reply:
I agree with @mloskot.